src/data_watcher.cpp

```diff
@@ -161,31 +166,133 @@ std::optional<std::vector<eventInfo>> DataWatcher::readEvents()
 bool DataWatcher::processEvent(const eventInfo& receivedEventInfo)
 {
     if ((std::get<eventMask>(receivedEventInfo) & IN_CLOSE_WRITE) != 0)
+    {
+        return processCloseWrite(receivedEventInfo);
+    }
+    else if ((std::get<eventMask>(receivedEventInfo) &
+              (IN_CREATE | IN_ISDIR)) != 0)
     {
         /**
-         * Either of the following case occured
-         * Case 1 : Files listed in the config and are already existing has been
-         *          modified
-         * Case 2 : Files got created inside a watching directory
+         * Handle the creation of directories inside a monitoring DIR
          */
-        lg2::debug("Processing an IN_CLOSE_WRITE for {PATH}", "PATH",
-                   _watchDescriptors.at(std::get<wd>(receivedEventInfo)) /
-                       std::get<basename>(receivedEventInfo));
+        return processCreate(receivedEventInfo);
+    }
+    return false;
+}
+
+bool DataWatcher::processCloseWrite(const eventInfo& receivedEventInfo)
+{
+    lg2::debug("Processing an IN_CLOSE_WRITE for {PATH}", "PATH",
+               _watchDescriptors.at(std::get<wd>(receivedEventInfo)) /
+                   std::get<basename>(receivedEventInfo));
+    if (_watchDescriptors.at(std::get<0>(receivedEventInfo))
+            .string()
+            .starts_with(_dataPathToWatch.string()))
+    {
+        // Case 1 : The file configured in the JSON exists and is modified
+        // Case 2 : A file got created or modified inside a watching subdir
         return true;
     }
-    else if ((std::get<eventMask>(receivedEventInfo) &
-              (IN_CREATE | IN_ISDIR)) != 0)
+    else if (_watchDescriptors.at(std::get<0>(receivedEventInfo)) /
```

Comment:
Use `using alias` for wd and basename. the same comment applicable to all places.